service/history/historyEngine.go

```diff
@@ -167,27 +167,13 @@ func (e *historyEngineImpl) Start() {
 	logging.LogHistoryEngineStartingEvent(e.logger)
 	defer logging.LogHistoryEngineStartedEvent(e.logger)
 
+	e.registerDomainFailoverCallback()
```

Comment:
You are setting up the callback before starting the processors.  This could be problematic if the callback is received before the processors are even started.

Reply:
the failover processor is independent of the normal processor